owasp-top10-2017-apps/a10/games-irados/app/routes.py

```diff
@@ -77,6 +78,12 @@ def login():
         psw = Password(request.form.get('password').encode('utf-8'))
         user_password, success = database.get_user_password(username)
         if not success or user_password == None or not psw.validate_password(str(user_password[0])):
+            logger.error(dict(
+                ts=str(datetime.datetime.now()),
+                action='login',
+                username=username,
```

Comment:
When an user tries to login mistakenly using his/her password as his/her username, your log will register this sensitive data:

```
172.30.0.1 - - [12/Mar/2019 17:40:07] "POST /login HTTP/1.1" 200 -
{'ts': '2019-03-12 17:40:34.642044', 'action': 'login', 'username': b'password123', 'message': 'Login failed'}
172.30.0.1 - - [12/Mar/2019 17:40:34] "POST /login HTTP/1.1" 200 -
```
Do you ming fixing this sensitive data exposure? 😃

Reply:
Good point. Done!